### PR TITLE
Allow to select an example on the front page

### PR DESCRIPTION
**Pull request by wilzbach**
Marking this as "needs work" as I will to rebase this once we agree on #1775

image

image

### CODE REVIEW

**Comment by wilzbach** [comment]
Alrighty, finally thanks to the minimal solution in #1775 this should be ready.

**Comment by CyberShadow** [comment]
First: Diff looks great! Thanks for not giving up.

Dropdown looks a bit weird for me, any idea why?

**Comment by wilzbach** [comment]
Okay, I think we can fix this by providing our own selector symbol from FontAwesome.
However, there are still UI differences:

Chrom(ium):

image

Firefox:

image

**Comment by CyberShadow** [comment]
Alright, but why is that necessary? Why isn't the browser rendering the arrow?

**Comment by wilzbach** [comment]
> Alright, but why is that necessary? Why isn't the browser rendering the arrow?

Without it looked really ugly in my Firefox:

image

**Comment by CyberShadow** [comment]
> Without it looked really ugly in my Firefox:

That's because we're setting the background color. If we exclude that block from having its color and background color set, it should use the native styling... with the downside that it will look a little weird to people who have their browsers set to have a light-on-dark color theme by default. I don't think we need to worry about that, as long as the website is not unusable to them.

**Comment by CyberShadow** [comment]
I guess either way works, but if possible, I think it would be better to not rely on font-awesome for the front page.

**Comment by wilzbach** [comment]
>  If we exclude that block from having its color and background color set, it should use the native styling... with the downside that it will look a little weird to people who have their browsers set to have a light-on-dark color theme by default. I don't think we need to worry about that, as long as the website is not unusable to them.

That wouldn't look good:

image

I actually tried to be close to styling of the "Install X" boxes.

> I guess either way works, but if possible, I think it would be better to not rely on font-awesome for the front page.

FWIW we already rely on it:

image

AFAICT it's the only way to ensure a consistent experience across different browsers.

**Comment by CyberShadow** [comment]
> That wouldn't look good:

No, that's not right. You're trying to undo the `background` rule with another `background` rule, right? That's not going to work. There must be no `background` rule in effect at all.

**Comment by CyberShadow** [comment]
> FWIW we already rely on it:

Ah, that's right.

**Comment by CyberShadow** [comment]
Seb do you want to try the no-CSS approach or should we go forward with the current FontAwesome version?

**Comment by wilzbach** [comment]
> Seb do you want to try the no-CSS approach or should we go forward with the current FontAwesome version?

Hmm, the no-CSS approach would look like this:

image

image

>  or should we go forward with the current FontAwesome version?

I am fine with both ways (slight preference on FontAwesome because it keeps the styling consistent).

**Comment by CyberShadow** [comment]
Needs some tweaking for mobile:

File: index.dd [hunk 1]
```diff
@@ -481,7 +504,11 @@ Macros:
     TAG2=<$1 $2>$3</$1>
     D=<span class="d_inlinecode">$0</span>
     EXAMPLE=$(TAG2 a, id="a$1-control" class="example-control", )$(TAG2 div, id="a$1" class="example-box", $(RUNNABLE_EXAMPLE $2))
-    EXTRA_EXAMPLE=<div class="your-code-here-extra" style="display:none">$(RUNNABLE_EXAMPLE $0)</div>
+    EXTRA_EXAMPLE=<div class="your-code-here-extra" style="display:none">
+        <div class="your-code-here-title">$1</div>
```
Comment (andralex) [inline index.dd]: DIVC?

**Comment by wilzbach** [comment]
> Needs some tweaking for mobile:

Thanks. I came to the conclusion that tip box hasn't been used [much](http://forum.dlang.org/search?q=%22%5Byour+code+here%5D%22&search=Search) over the last years and especially users on their phone most likely won't post an example, so imho the best solution is also the simplest here: hiding the tip box on devices with a narrow layout.

**Comment by CyberShadow** [comment]
Still broken for very narrow widths:

dtest dlang io-artifact-website-ec00ac7dd35faa307819ad634ab38bf91aabacb4-3f759d90a402ac3595f96e341659d3c6-web-index html

**Comment by wilzbach** [comment]
> Still broken for very narrow widths:

I can't really reproduce this in Chrome or FF, but I added `max-width: 100%` which should fix things even for these tiny layouts:

image

@CyberShadow does it work for you now?

**Comment by CyberShadow** [comment]
> @CyberShadow does it work for you now?

Looks perfect, thanks!

File: index.dd [hunk 2]
```diff
@@ -110,6 +108,31 @@ $(SCRIPT
...
+        el.value = i;
+        el.innerText = title;
+        sel.appendChild(el);
+    });
+    sel.selectedIndex = rouletteIndex;
+    sel.addEventListener("change", function(e) {
+        examples[rouletteIndex].style.display = "none";
+        rouletteIndex = sel.selectedIndex;
+        examples[rouletteIndex].style.display = "block";
+    });
```
Comment (CyberShadow) [inline index.dd]: The event handler could have been added later in the main JS file, but it's just 3 lines so no matter.

**Comment by CyberShadow** [comment]
> tip box hasn't been used much over the last years

Forgot to post this earlier, but I think you simply found a bug in the forum search (or SQLite indexing). I know I fixed one yesterday (regarding deleted posts), but that search and its variations is producing inconsistent results with some variations of it (for one, the query needs to search the subject, not body).

**Pull request merged**